telegram-bot-api/StatsJson.h

```diff
@@ -0,0 +1,285 @@
+    object("resident_size", JsonStatsSize(mem_stat_.resident_size_));
+    object("resident_size_peak", JsonStatsSize(mem_stat_.resident_size_peak_));
+    object("virtual_size", JsonStatsSize(mem_stat_.virtual_size_));
+    object("virtual_size_peak", JsonStatsSize(mem_stat_.virtual_size_peak_));
+  }
+
+ private:
+  const td::MemStat mem_stat_;
+};
+
+
+class JsonStatsCpuItem : public td::Jsonable {
+ public:
+  JsonStatsCpuItem() : total_cpu_("unset"), user_cpu_("unset"), system_cpu_("unset") {
+  }
+  JsonStatsCpuItem(const td::string &total_cpu, const td::string user_cpu, const td::string system_cpu)
+      : total_cpu_(total_cpu), user_cpu_(user_cpu), system_cpu_(system_cpu) {
+  }
+  void store(td::JsonValueScope *scope) const {
+    auto object = scope->enter_object();
+    // Maybe needs td::JsonString(td::Slice(...)) instead of just a string?
+    object("total_cpu", total_cpu_);
+    object("user_cpu", user_cpu_);
+    object("system_cpu", system_cpu_);
+  }
+
+  td::string total_cpu_;
+  td::string user_cpu_;
+  td::string system_cpu_;
+ private:
+};
+
+class JsonStatsCpuStat : public td::Jsonable {
+ public:
+  explicit JsonStatsCpuStat(const StatItem& inf, const StatItem& five_sec, const StatItem& one_min, const StatItem& one_hour) :
+      inf_(inf), five_sec_(five_sec), one_min_(one_min), one_hour_(one_hour) {
+  }
+  void store(td::JsonValueScope *scope) const {
+    auto object = scope->enter_object();
+    object(td::Slice(ServerCpuStat::DESCR[0]), td::JsonString(td::Slice(inf_.value_)));
+    object(td::Slice(ServerCpuStat::DESCR[1]), td::JsonString(td::Slice(five_sec_.value_)));
+    object(td::Slice(ServerCpuStat::DESCR[2]), td::JsonString(td::Slice(one_min_.value_)));
+    object(td::Slice(ServerCpuStat::DESCR[3]), td::JsonString(td::Slice(one_hour_.value_)));
+  }
+ private:
+  const StatItem& inf_;
+  const StatItem& five_sec_;
+  const StatItem& one_min_;
+  const StatItem& one_hour_;
+};
+
+class JsonStatsCpu : public td::Jsonable {
+ public:
+  explicit JsonStatsCpu(td::vector<td::vector<StatItem>> cpu_stats) : cpu_stats_(std::move(cpu_stats)) {
+  }
+  void store(td::JsonValueScope *scope) const {
+    auto object = scope->enter_object();
+    CHECK(cpu_stats_.size() == ServerCpuStat::SIZE);
+    CHECK(cpu_stats_[0].size() == 3);
+    CHECK(cpu_stats_[1].size() == 3);
+    CHECK(cpu_stats_[2].size() == 3);
+    object("total_cpu", JsonStatsCpuStat(cpu_stats_[0][0], cpu_stats_[1][0], cpu_stats_[2][0], cpu_stats_[3][0]));
+    object("user_cpu", JsonStatsCpuStat(cpu_stats_[0][1], cpu_stats_[1][1], cpu_stats_[2][1], cpu_stats_[3][1]));
+    object("system_cpu", JsonStatsCpuStat(cpu_stats_[0][2], cpu_stats_[1][2], cpu_stats_[2][2], cpu_stats_[3][2]));
+  }
+
+ private:
+  const td::vector<td::vector<StatItem>> cpu_stats_;
+};
+
+class JsonStatsBot : public td::Jsonable {
+ public:
+  explicit JsonStatsBot(const std::pair<td::int64, td::uint64> score_id_pair) : score_id_pair_(score_id_pair) {
```

Comment:
use std::move in score_id_pair as you are already moving it from ClientManager when instantiating JsonStatsBotAdvanced

Reply:
Done